build/patches/Always-ask-for-permissions.patch

```diff
@@ -0,0 +1,291 @@
++++ b/chrome/android/java/src/org/chromium/chrome/browser/infobar/PermissionUpdateInfoBarDelegate.java
+@@ -23,6 +23,11 @@ import java.util.Collections;
+ import java.util.HashSet;
+ import java.util.Set;
+ 
++import org.chromium.components.permissions.AndroidPermissionRequester;
++
++// remove this comment when moved to
++// chrome/android/java/src/org/chromium/chrome/browser/permissions/PermissionUpdateRequester.java
++
+ /**
+  * Handles requesting the android runtime permissions for the permission update infobar.
+  */
+@@ -131,7 +136,10 @@ class PermissionUpdateInfoBarDelegate implements PermissionCallback {
+                 if (!mRequiredAndroidPermissions.contains(mAndroidPermisisons[i])) {
+                     continue;
+                 }
+-                hasAllPermissions &= windowAndroid.hasPermission(mAndroidPermisisons[i]);
++                boolean hasPermission = windowAndroid.hasPermission(mAndroidPermisisons[i]);
++                hasAllPermissions &= hasPermission;
++                if (hasPermission)
++                    AndroidPermissionRequester.setUserGrantedPermission(mAndroidPermisisons[i]);
+             }
+         }
+         if (mNativePtr != 0) {
+diff --git a/chrome/browser/permissions/permission_update_infobar_delegate_android.cc b/chrome/browser/permissions/permission_update_infobar_delegate_android.cc
+--- a/chrome/browser/permissions/permission_update_infobar_delegate_android.cc
++++ b/chrome/browser/permissions/permission_update_infobar_delegate_android.cc
+@@ -53,6 +53,9 @@ infobars::InfoBar* PermissionUpdateInfoBarDelegate::Create(
+     permissions::AppendOptionalAndroidPermissionsForContentSetting(
+         content_settings_type, &optional_permissions);
+ 
++    // in bromite there are other contentsettings defined as permission
++    // so in debug please don't crash
++    // moved to chrome/browser/permissions/permission_update_message_controller_android.cc
+     if (message_id == -1) {
+       if (content_settings_type == ContentSettingsType::GEOLOCATION) {
+         message_id = IDS_INFOBAR_MISSING_LOCATION_PERMISSION_TEXT;
+@@ -65,16 +68,16 @@ infobars::InfoBar* PermissionUpdateInfoBarDelegate::Create(
+       } else if (content_settings_type == ContentSettingsType::AR) {
+         message_id = IDS_INFOBAR_MISSING_AR_CAMERA_PERMISSION_TEXT;
+       } else {
+-        NOTREACHED();
++        //NOTREACHED();
+       }
+     } else if (message_id == IDS_INFOBAR_MISSING_CAMERA_PERMISSION_TEXT) {
+-      DCHECK(content_settings_type == ContentSettingsType::MEDIASTREAM_MIC);
++      //DCHECK(content_settings_type == ContentSettingsType::MEDIASTREAM_MIC);
+       message_id = IDS_INFOBAR_MISSING_MICROPHONE_CAMERA_PERMISSIONS_TEXT;
+     } else if (message_id == IDS_INFOBAR_MISSING_MICROPHONE_PERMISSION_TEXT) {
+-      DCHECK(content_settings_type == ContentSettingsType::MEDIASTREAM_CAMERA);
++      //DCHECK(content_settings_type == ContentSettingsType::MEDIASTREAM_CAMERA);
```

Comment:
These changes should also be removed from here

Reply:
do you intend not to comment or not to remove?
I put them because without those, I would not be able to use the debug version with active dchecks, it would often crash. there are already other parts that make me crash bromite in debug: every new version I have to remove them, I would not like to have others.

Reply:
> there are already other parts that make me crash bromite in debug: every new version I have to remove them, I would not like to have others.

The solution is to make the DCHECKs not fail, not to ignore them. For example the `NOTREACHED()`: we cannot just ignore that.